src/Arcus.Security.Providers.AzureKeyVault/Authentication/ManagedServiceIdentityAuthentication.cs

```diff
@@ -37,20 +55,33 @@ public ManagedServiceIdentityAuthentication(string connectionString = null, stri
         /// <returns>A <see cref="IKeyVaultClient" /> client to use for interaction with the vault</returns>
         public Task<IKeyVaultClient> AuthenticateAsync()
```

Comment:
While reviewing this PR I'm thinking if we shouldn't have a `KeyVaultAuthentication` with this method and maybe the basis for AuthenticateClient?

(in separate PR of course)

Reply:
Yes, was something I proposed a while ago too, maybe we can reuse it even in the invictus framework?

Reply:
I think we should look at adding support for https://docs.microsoft.com/en-us/dotnet/azure/authentication as well.

Reply:
Follow-up PR. Resolving.